Sources/StreamChatUI/ChatMessageList/ChatMessageListKeyboardObserver.swift

```diff
@@ -40,58 +40,29 @@ open class ChatMessageListKeyboardObserver {
     private func keyboardWillChangeFrame(_ notification: Notification) {
         guard
             viewController?.presentedViewController == nil,
-            let frame = (notification.userInfo?[UIResponder.keyboardFrameEndUserInfoKey] as? NSValue)?.cgRectValue,
-            let oldFrame = (notification.userInfo?[UIResponder.keyboardFrameBeginUserInfoKey] as? NSValue)?.cgRectValue,
-            let duration = notification.userInfo?[UIResponder.keyboardAnimationDurationUserInfoKey] as? Double,
-            let curve = notification.userInfo?[UIResponder.keyboardAnimationCurveUserInfoKey] as? UInt
+            let userInfo = notification.userInfo,
+            let keyboardFrame = userInfo[UIResponder.keyboardFrameEndUserInfoKey] as? CGRect,
+            let keyboardAnimationDuration = userInfo[UIResponder.keyboardAnimationDurationUserInfoKey] as? TimeInterval,
+            let keyboardAnimationCurve = userInfo[UIResponder.keyboardAnimationCurveUserInfoKey] as? UInt
         else { return }
 
-        let localFrame = containerView.convert(frame, from: nil)
-        let localOldFrame = containerView.convert(oldFrame, from: nil)
+        let convertedKeyboardFrame = containerView.convert(keyboardFrame, from: UIScreen.main.coordinateSpace)
+        let intersectedKeyboardHeight = containerView.frame.intersection(convertedKeyboardFrame).height
 
-        // message composer follows keyboard
-        composerBottomConstraint?.constant = -(containerView.bounds.height - localFrame.minY)
+        composerBottomConstraint?.constant = -intersectedKeyboardHeight
 
-        // calculate new contentOffset for message list, so bottom message still visible when keyboard appears
-        var keyboardTop = localFrame.minY
-        if keyboardTop == containerView.bounds.height {
-            keyboardTop -= containerView.safeAreaInsets.bottom
-        }
+        let indexPathNearestToKeyboard = collectionView.indexPathsForVisibleItems.sorted().first
```

Comment:
Ahah yes, we could play with the contentOffset.y, but that requires more calculations when the collection view is already scrolled to the bottom. This solution is simpler and IMO better, since the cells don't get cut in half. (Although on other apps, this happens)